rfcs/001-foundational-architecture.md

```diff
@@ -0,0 +1,564 @@
+**Pros**
+
+- Whether MessageFormat 2.0, Fluent, or something else becomes the standard does not matter.
+
+**Cons**
+
+- Engineering complexity to support numerous syntaxes.
+
+- Inferior user experience "what syntax supports what?".
+
+- There should not be 10+ different syntaxes to express human languages.
+
+### SDK
+
+Messages (strings) need to be retrieved and formatted. That's the job of an i18n SDK. Most implementations make use of a key-value resource and a lookup function called `t` (translate), or a translation component. From a developer perspective, the i18n SDK loads resources, detects the language of a user and formats the message. In other words: "The message `example` should be displayed here in the correct language and format for me".
+
+#### Illustration
+
+> Note: The following is a high level overview. Precise implementations and features vary.
+
+```mermaid
+flowchart LR
+    subgraph Resources
+      direction LR
+      EN
+      DE
+    end
+
+    subgraph SDK
+        direction LR
+        Language[Language detection]
+        Formatting
+    end
+
+    Resources-->SDK-->Output
+
+```
+
+##### Resources
+
+```json
+{
+  "example": "Welcome {$name} to this example.",
+  "info": "Click the button to continue."
+}
+```
+
+##### Retrieving a message
+
+```jsx
+import translate from "sdk";
+
+console.log(translate("example", { name: "Samuel" }));
+```
+
+##### Output
+
+```
+>> Welcome Samuel to this example.
+```
+
+#### Observations
+
+1. A variety of good and adopted open source SDKs exist [[1](https://github.com/ivanhofer/typesafe-i18n), [2](https://github.com/formatjs/formatjs), [3](https://pub.dev/packages/flutter_i18n), [4](https://github.com/solidjs-community/solid-primitives/tree/main/packages/i18n), ...]. Each serves a different programming language, framework, niche, or feature.
+
+2. The internals are identical:  
+   `Resource` -> `Reference and Format a Message` -> `Output`
+
+#### Proposal
+
+Leverage existing SDKs instead of forcing a specific inlang SDK.
+
+**Pros**
+
+- Existing codebases can adopt inlang.
+- Different SDKs solve different problems.
+
+**Cons**
+
+- The user experience could potentially be higher with a dedicated SDK.
+
+#### Considered alternative
+
+Develop (and require) a dedicated SDK developed by inlang.
+
+**Pros**
+
+- Potentially better user experience by delivering an "end to end" solution.
+
+**Cons**
+
+- Adoption is severely limited.
+- Different SDKs exist because a "one size fits all" solution likely does not work.
+
+### Dev tools
+
+Using an SDK does not relieve developers from two extremely time-consuming and ever-repeating tasks:
+
+1. Extracting hard-coded strings.
+
+   Developers have to manually copy & paste hardcoded strings into resources. That process can take weeks.
+
+2. Validating translations.
+
+   No widely adopted tools exist to validate translations: Does the German translation exist? Is the French translation correct? Does the UI work as expected for Arabic?
```

Comment:
'Is the french translation correct' seems like a poor example. Sounds like we are in the business of translating strings rather than providing tools for localization.

Reply:
Changed "translation" to "message".